apps/root/src/common/components/token-amount-input/index.tsx

```diff
@@ -65,50 +55,121 @@ const TokenAmountInput = ({
+      <ContainerBox>
+        <ContainerBox flexDirection="column" gap={2} alignItems="flex-start" justifyContent="center">
+          <Typography variant="body1">{label}</Typography>
+          <TokenPickerButton
+            disabled={isLoadingRoute}
+            token={token}
+            showAction
+            onClick={() => startSelectingCoin(selectedToken || emptyTokenWithAddress('token'))}
+          />
+          {!isUndefined(balance) && token && (
+            <ContainerBox alignItems="center" gap={1}>
+              <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+                <EmptyWalletIcon />
+              </Typography>
+              <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+                {isLoadingBalance ? (
+                  <Skeleton variant="text" sx={{ minWidth: '10ch' }} />
+                ) : (
+                  <>
+                    {balance.amountInUnits}
+                    {balance.amountInUSD && ` / ≈$${balance.amountInUSD}`}
+                  </>
+                )}
+              </Typography>
+            </ContainerBox>
+          )}
+        </ContainerBox>
+        <ContainerBox flexDirection="column" flex="1" alignItems="flex-end">
+          {maxBalanceBtn && !isUndefined(balance) && selectedToken && (
+            <Button
+              onClick={onSetMaxBalance}
+              disabled={isLoadingRoute}
+              color="primary"
+              variant="text"
+              sx={{ padding: 0, minWidth: '10px', marginTop: SPACING(2) }}
+            >
+              <FormattedMessage description="maxWallet" defaultMessage="Max" />
+            </Button>
+          )}
+          <ContainerBox flexDirection="column" flex="1" alignItems="flex-end" justifyContent="center">
+            <FormControl variant="standard" fullWidth>
+              <Input
+                id={id}
+                onChange={(evt) =>
+                  amountValidator({
+                    onChange: onSetTokenAmount,
+                    nextValue: evt.target.value,
+                    decimals: token?.decimals || 18,
+                  })
+                }
+                value={tokenAmount.amountInUnits}
+                onFocus={() => setIsFocused(true)}
+                onBlur={() => setIsFocused(false)}
+                autoComplete="off"
+                placeholder="0"
+                disableUnderline
+                inputProps={{ style: { textAlign: 'right', height: 'auto' } }}
+                sx={{ ...buildTypographyVariant(mode).h3, fontWeight: '700', color: 'inherit', textAlign: 'right' }}
+              />
+            </FormControl>
+            <ContainerBox>
+              <Typography variant="body">≈{` $${tokenAmount.amountInUSD || '0'}`}</Typography>
+              {priceImpact && !isNaN(Number(priceImpact)) && isFinite(Number(priceImpact)) && (
```

Comment:
We should add a gap={1} between priceImpact and amountInUSD. Also the color for priceImpact doesn't adjust to light theme

<img width="579" alt="Screenshot 2024-03-04 at 17 16 15" src="https://github.com/Mean-Finance/dca-fe/assets/144699868/bceb546d-d843-48f1-9dad-2be71523b3c2">